lib/sidekiq/instrument/mixin.rb

```diff
@@ -13,10 +13,17 @@ def worker_dog_options(worker)
     end
 
     def max_retries(worker)
-      retries = worker.class.get_sidekiq_options['retry'] || Sidekiq[:max_retries]
```

Comment:
This was causing the worker to always default to the global `Sidekiq[:max_retries]` value even if`worker.class.get_sidekiq_options['retry']worker.class.get_sidekiq_options['retry']` was set to false, which is NOT the intended behavior.